File: web-api/src/models/regulator/regulator_parameters_model.py
from models.abstracts.app_base_model import AppBaseModel

from models.regulator.control_parameters_model import ControlParametersModel
from models.regulator.regulation_parameters_model import RegulationParametersModel
from models.regulator.schedules_model import SchedulesModel
from models.regulator.temperature_graph_model import TemperatureGraphModel


# Параметры
class RegulatorParametersModel(AppBaseModel):
    # Управление
    control_parameters: ControlParametersModel

    # Темп. график
    temperature_graph: TemperatureGraphModel

    # Расписание
    schedules: SchedulesModel

    # Регулятор
    regulation_parameters: RegulationParametersModel

File: web-ui/src/models/regulator-settings/regulator-parameters-model.ts
import { ControlParametersModel } from './control-parameters-model';
import { RegulationParametersModel } from './regulation-parameters-model';
import { TemperatureGraphModel } from './temperature-graph-model';
import { SchedulesModel } from './schedules-model'

export type RegulatorParametersModel = {
    controlParameters: ControlParametersModel;

    temperatureGraph: TemperatureGraphModel;

    schedules: SchedulesModel;

    regulationParameters: RegulationParametersModel;
}